infrastructure/bls-keystore/src/main/java/tech/pegasys/teku/bls/keystore/KeyStoreLoader.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright ConsenSys Software Inc., 2020
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.bls.keystore;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidTypeIdException;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.bls.keystore.model.KdfParam;
+import tech.pegasys.teku.bls.keystore.model.KeyStoreData;
+
+/** Provide utility methods to load/store BLS KeyStore from json format */
+public class KeyStoreLoader {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper()
+          .registerModule(new KeyStoreBytesModule())
+          .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  public static KeyStoreData loadFromString(final String keystoreString) {
+    try {
+      final KeyStoreData keyStoreData = OBJECT_MAPPER.readValue(keystoreString, KeyStoreData.class);
+      keyStoreData.validate();
+      return keyStoreData;
+    } catch (final JsonParseException e) {
+      throw new KeyStoreValidationException("Invalid KeyStore: " + e.getMessage(), e);
+    } catch (final JsonMappingException e) {
+      throw convertToKeyStoreValidationException(e);
+    } catch (final IOException e) {
+      LOG.error("Unexpected IO error while reading KeyStore: " + e.getMessage());
+      throw new KeyStoreValidationException(
+          "Unexpected IO error while reading KeyStore: " + e.getMessage(), e);
+    }
+  }
+
+  public static KeyStoreData loadFromFile(final URI keystoreFile)
```

Comment:
Curious to know the reason for using a URI instead a Path to the file. Either will work fine though.

Reply:
seemed a lot more successful running windows tests, we were getting a bunch of resource errors, and we use URL most places rather than path -> string -> path -> file

Reply:
I had thought we were just using it and changing interfaces wouldnt matter, but i can probably try to come up with a different solution if this is a problem...

Reply:
Don't think it's a problem, should work fine with that. Was more just curious.